docs/core/context.md

```diff
@@ -0,0 +1,101 @@
+- **Gas Meters:** Specifically, a `gasMeter` for the transaction currently being processed using the context and a `blockGasMeter` for the entire block it belongs to. Users specify how much in fees they wish to pay for the execution of their transaction; these gas meters keep track of how much [gas](../basics/accounts-fees-gas.md) has been used in the transaction or block so far. If the gas meter runs out, execution halts.
+- **CheckTx Mode:** A boolean value indicating whether a transaction should be processed in `CheckTx` or `DeliverTx` mode.
+- **Min Gas Price:** The minimum [gas](../basics/accounts-fees-gas.md) price a node is willing to take in order to include a transaction in its block. This price is a local value configured by each node individually.
+- **Consensus Params:** The ABCI type [Consensus Parameters](https://tendermint.com/docs/spec/abci/apps.html#consensus-parameters), which enforce certain limits for the blockchain, such as maximum gas for a block.
+- **Event Manager:** An event manager with a number of [`Event`](https://github.com/cosmos/cosmos-sdk/blob/master/types/events.go) objects that can be emitted from. Modules developers may define module `events` and `attributes` or use the SDK types, and emit an `EventTypeMessage` whenever a message is handled. For example:
+```go
+ctx.EventManager().EmitEvent(sdk.NewEvent(
+    sdk.EventTypeMessage,
+    sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),))
+```
+
+## Go Context Package
+
+A basic Context is defined in the [Golang Context Package](https://golang.org/pkg/context). A Context is an immutable data structure that carries request-scoped data across APIs and processes. Contexts are also designed to enable concurrency and to be used in goroutines.
+
+Contexts are intended to be **immutable**. They should never be edited. Instead, the convention is to create a child context from its parent using a `With` function. For example:
+
+``` go
+childCtx = parentCtx.WithBlockHeader(header)
+```
+
+The [Golang Context Package]((https://golang.org/pkg/context) documentation instructs developers to explicitly pass a context `ctx` as the first argument of a process.
+
+## Cache Wrapping
+
+The Context contains a Multistore, which allows for cache wrapping functionality: a deep copy of the state stored in a Multistore can be made. Processes are free to write changes to the copy of the state, then write the changes back to the original state or disregard them if something goes wrong. The pattern of usage for a Context is as follows:
+
+1. A process receives a Context `ctx` from its parent process, which provides information needed to perform the process.
+2. The `ctx` is [**cache wrapped**](./multistore.md), i.e. a copy of the [multistore](./multistore.md) is made so that the process can make changes to the state as it executes, without changing the original `ctx`.
```

Comment:
```suggestion
2. The `ctx.ms` is [**cache wrapped**](./multistore.md), i.e. a cached copy of the [multistore](./multistore.md) is made so that the process can make changes to the state as it executes, without changing the original `ctx.ms` state.
```